mods/d2k/rules/defaults.yaml

```diff
@@ -499,6 +500,7 @@
 		Sequence: invisible
 	Sellable:
 		RequiresCondition: !being-demolished
```

Comment:
Little scope-creep: Shouldn't this be `!build-incomplete && !being-demolished` like on `^Building` and could then be removed here? Or is there something special about D2k defences?

Reply:
I don't think the `!build-incomplete` is necessary as walls are instantly created.

Reply:
`medium_gun_turret` has a make animation and inherits `^Defense`. Could be a case of "bug or feature", I haven't tested ingame.

Reply:
This was intentionally set up like this in https://github.com/OpenRA/OpenRA/commit/94088d37a63891246abe62b7fe6c75a09ad12464#diff-dc897ec3e7180a07e0129e4982c06f13c32a9c163056127e085cec857305ee6e by @pchote.

Reply:
That was long before #17817 added make animations. Missing that condition now probably causes bugs.

Reply:
I think this can also be done in a follow-up if @Mailaender doesn't want to drag this PR out further.

Reply:
Indeed, thats what I meant by saying it may also need to be fixed on prep.

Reply:
As only turrets inherit `^Defense` I added the fix.

Reply:
Can you please split that into its own PR that we can track for prep?

Reply:
You can cherry pick that into prep.

Reply:
Or take both, it is a really trivial change.

Reply:
It is and I don't want to be annoying... but can't `Sellable` not be removed from `^Defense` (as it is inheriting `^Building`)? Not that it made any difference...

Reply:
No, you're right. This is redundant, and defenses do use `SellSounds: BUILD1.WAV` (which is inherited from ^Building).

Reply:
Updated this and tested that walls still build properly.